UnifiedPDF: Add instructions and icon to unlock UI

hortont424 opened the pull request:
#### 77ac3375317c66a06fc0cd93c216f5295d73c4ab
<pre>
UnifiedPDF: Add instructions and icon to unlock UI
<a href="https://bugs.webkit.org/show_bug.cgi?id=269345">https://bugs.webkit.org/show_bug.cgi?id=269345</a>
<a href="https://rdar.apple.com/122929864">rdar://122929864</a>

Reviewed by Simon Fraser.

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
Export XMLNSNames.h.

* Source/WebCore/en.lproj/Localizable.strings:
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::pdfPasswordFormTitle):
(WebCore::pdfPasswordFormSubtitle):
(WebCore::pdfPasswordFormInvalidPasswordSubtitle):
* Source/WebCore/platform/LocalizedStrings.h:
Add some strings.

* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginAnnotation.mm:
(WebKit::PDFPluginAnnotation::attach):
If a PDFPluginAnnotation subclass puts a class on its element, don&apos;t overwrite it.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
Update our inline style and add some new rules for the password form.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginPasswordField.h:
(WebKit::PDFPluginPasswordField::PDFPluginPasswordField):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginPasswordField.mm:
(WebKit::PDFPluginPasswordField::resetField):
Add a method to clear the field.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginTextAnnotation.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginTextAnnotation.mm:
(WebKit::PDFPluginTextAnnotation::createAnnotationElement):
(WebKit::PDFPluginTextAnnotation::setValue):
Factor out setValue (and implement it in a simpler way).

* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::createPasswordEntryForm):
Install the form (UnifiedPDF only) ahead of the field (used by both plugins).

(WebKit::UnifiedPDFPlugin::attemptToUnlockPDF):
If the unlock fails, clear the field and change the form&apos;s instructions to note that.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginPasswordForm.h: Added.
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginPasswordForm.mm: Added.
Add PDFPluginPasswordForm, which vends the UI above the password entry field
for UnifiedPDF.

Canonical link: <a href="https://commits.webkit.org/274609@main">https://commits.webkit.org/274609@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/940aa070ef6a356b8a31abe86db1a6cfa4a4715e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/39589 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/18568 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/41945 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/42124 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/35489 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/41895 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/21458 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/15897 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/33055 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/40163 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/15656 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/34259 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/13576 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/13558 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/35212 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/43401 "Built successfully") | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/33042 "Build is in progress. Recent messages:") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/35946 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/35555 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/39337 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/14401 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/11859 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/37600 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/39636 "Build is in progress. Recent messages:") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/16007 "Built successfully") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/46221 "Build is in progress. Recent messages:") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8859 "Built successfully and passed tests") | [❌ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/16055 "Failed to checkout and rebase branch from PR 24397") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/35/builds/46221 "Build is in progress. Recent messages:") | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/15664 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebCore/en.lproj/Localizable.strings:
```diff
@@ -1453,6 +1456,9 @@
 "This credential is malformed" = "This credential is malformed";
 
+/* Title when a PDF needs a password to be unlocked */
+"This document is password protected." = "This document is password protected.";
```
"password protected" or "password-protected"?

hortont424 replied:
This matches preview.

smfr commented on Source/WebKit/WebProcess/Plugins/PDF/PDFPluginAnnotation.mm:
```diff
@@ -72,7 +72,8 @@
     m_element = element.copyRef();
 
-    element->setAttributeWithoutSynchronization(classAttr, "annotation"_s);
+    if (!element->hasClass())
```
Unclear what this is doing.

hortont424 replied:
That's why I explained in the commit message.

smfr commented on Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
```diff
@@ -356,31 +356,60 @@ class PDFPluginBase : public ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr<PDF
+    "    right: 0;"
+    "    bottom: 0;"
+    "    display: flex;"
+    "    flex-direction: column;"
```
Yay modern flex

smfr commented on Source/WebKit/WebProcess/Plugins/PDF/PDFPluginPasswordForm.mm:
```diff
@@ -0,0 +1,92 @@
+    return adoptRef(*new PDFPluginPasswordForm(plugin));
+}
+
+PDFPluginPasswordForm::~PDFPluginPasswordForm()
```
`= default`

hortont424 committed:
UnifiedPDF: Add instructions and icon to unlock UI
https://bugs.webkit.org/show_bug.cgi?id=269345
rdar://122929864

Reviewed by Simon Fraser.

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
Export XMLNSNames.h.

* Source/WebCore/en.lproj/Localizable.strings:
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::pdfPasswordFormTitle):
(WebCore::pdfPasswordFormSubtitle):
(WebCore::pdfPasswordFormInvalidPasswordSubtitle):
* Source/WebCore/platform/LocalizedStrings.h:
Add some strings.

* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginAnnotation.mm:
(WebKit::PDFPluginAnnotation::attach):
If a PDFPluginAnnotation subclass puts a class on its element, don't overwrite it.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
Update our inline style and add some new rules for the password form.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginPasswordField.h:
(WebKit::PDFPluginPasswordField::PDFPluginPasswordField):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginPasswordField.mm:
(WebKit::PDFPluginPasswordField::resetField):
Add a method to clear the field.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginTextAnnotation.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginTextAnnotation.mm:
(WebKit::PDFPluginTextAnnotation::createAnnotationElement):
(WebKit::PDFPluginTextAnnotation::setValue):
Factor out setValue (and implement it in a simpler way).

* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::createPasswordEntryForm):
Install the form (UnifiedPDF only) ahead of the field (used by both plugins).

(WebKit::UnifiedPDFPlugin::attemptToUnlockPDF):
If the unlock fails, clear the field and change the form's instructions to note that.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginPasswordForm.h: Added.
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginPasswordForm.mm: Added.
Add PDFPluginPasswordForm, which vends the UI above the password entry field
for UnifiedPDF.

Canonical link: https://commits.webkit.org/274609@main
